static/scripts/assignment_table.js

```diff
@@ -0,0 +1,490 @@
+            data: data,
+            fixedColumnsLeft: lenFixedHeaders,
+            fixedRowsTop: lenFixedRowsText,
+            manualColumnMove: true,
+            manualColumnResize: true,
+            manualRowResize: true,
+            mergeCells: mergeCellsSettings,
+            hiddenColumns: {
+                columns: [0],
+                indicators: true,
+            },
+            hiddenRows: {
+                rows: [0],
+                indicators: true,
+            },
+            contextMenu: ['commentsAddEdit', 'commentsRemove', 'hidden_columns_hide', 'hidden_rows_hide', 'hidden_columns_show', 'hidden_rows_show'],
+            comments: true,
+            filters: true,
+            dropdownMenu: ['filter_by_value', 'filter_action_bar', 'undo'],
+            className: 'controlsQuickFilter htCenter htMiddle',
+            colHeaders: allHeaders,
+            columns: columns,
+            colWidths: 100,
+            columnHeaderHeight: 225,
+            rowHeaders: fixedRowsText,
+            rowHeaderWidth: 125,
+            // Disable the cells that contain user preferences
+            cells: function (row, col) {
+                const cellProperties = {};
+                if (row >= lenFixedRowsText && col >= lenFixedHeaders && row % 2 === 0) {
+                    cellProperties.readOnly = true;
+                }
+                return cellProperties;
+            },
+            afterInit: function () {
+                const sourceData = this.getSourceData();
+                saved_data.forEach(assignment => {
+                    const rowIndex = sourceData.findIndex(row => row.researchers.id === assignment.researcher_id);
+
+                    if (rowIndex !== -1) {
+                        const colIndex = this.propToCol(assignment.course_id);
+
+                        const commentsPlugin = this.getPlugin('comments');
+                        //Id is set for the two merged rows, and we want to set the comment for the assignment row
+                        commentsPlugin.setCommentAtCell(rowIndex + 1, colIndex, assignment.comment);
+                    }
+                });
+            },
+            afterGetColHeader: function (col, th) {
+                th.style.transform = 'rotate(180deg)';
+                th.style.writingMode = 'vertical-lr';
+                th.style.textAlign = 'center';
+                th.style.fontWeight = 'bold';
+                th.style.height = '100px';
+                th.style.lineHeight = '100px';
+
+                let colData = this.getDataAtCol(col);
+                if (col > lenFixedHeaders - 1) {
+                    if (colData[RowIndices.ASSISTANTS] === 0) {
+                        th.style.backgroundColor = '#E1BEE7';
+                    }
+                    else if (colData[RowIndices.TOTAL_ASSISTANT_NOW] >= colData[RowIndices.ASSISTANTS]) {
+                        th.style.backgroundColor = 'green';
+                    }
+                }
```

Comment:
Where is the red gone ? 

Now it is white when not enough researchers are assigned and turns green when enough are assigned. But if too many are assigned the titles does not change to red... Also the red color is missing from the legend.

Reply:
Sorry, it's my fault, I misunderstood your last comment.